Getting error when upgrading dokku

Hello all,

I installed dokku a while ago, wanted to upgrade from v0.31 to latest. I used the `boostrap.sh`, so wasn't sure if it installed deb version. I didn't find any deb packages installed on Ubuntu. I found my dokku git clone in /root/dokku and though I can follow the [source upgrading steps here](https://dokku.com/docs/getting-started/upgrading/?h=#upgrading-from-source).

Unfortunately, I'm getting an error. I don't understand what it means. Something about go. Any help would be appreciated.

Shell output:

```shell
root@app01 ~ # cd ~/dokku
root@app01 ~/dokku # git pull --tags origin master
From https://github.com/dokku/dokku
 * branch                master     -> FETCH_HEAD
Already up to date.
root@app01 ~/dokku # sudo DOKKU_BRANCH=master make install
apt-get update -qq
wget -qO /usr/local/bin/docker-image-labeler https://github.com/dokku/docker-image-labeler/releases/download/v0.8.0/docker-image-labeler-linux-amd64
chmod +x /usr/local/bin/docker-image-labeler
wget -qO /usr/local/bin/docker-container-healthchecker https://github.com/dokku/docker-container-healthchecker/releases/download/v0.11.0/docker-container-healthchecker-linux-amd64
chmod +x /usr/local/bin/docker-container-healthchecker
wget -qO /usr/local/bin/lambda-builder https://github.com/dokku/lambda-builder/releases/download/v0.8.0/lambda-builder-linux-amd64
chmod +x /usr/local/bin/lambda-builder
wget -qO /usr/local/bin/netrc https://github.com/dokku/netrc/releases/download/v0.10.0/netrc-linux-amd64
chmod +x /usr/local/bin/netrc
wget -qO /usr/local/bin/sshcommand https://github.com/dokku/sshcommand/releases/download/v0.18.0/sshcommand
chmod +x /usr/local/bin/sshcommand
sshcommand create dokku /usr/local/bin/dokku
wget -qO /usr/local/bin/plugn https://github.com/dokku/plugn/releases/download/v0.15.3/plugn-linux-amd64
chmod +x /usr/local/bin/plugn
wget -qO /usr/local/bin/procfile-util https://github.com/dokku/procfile-util/releases/download/v0.19.0/procfile-util-linux-amd64
chmod +x /usr/local/bin/procfile-util
apt-get -qq -y --no-install-recommends install curl
grep -i -E "^docker" /etc/group || groupadd docker
docker:x:999:dokku,ubuntu
usermod -aG docker dokku
wget -nv -O - https://get.docker.com/ | sh
2024-07-28 22:41:35 URL:https://get.docker.com/ [21582/21582] -> "-" [1]
# Executing docker install script, commit: 1ce4e39c9502b89728cdd4790a8c3895709e358d
Warning: the "docker" command appears to already exist on this system.

If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.

If you installed the current Docker package using this script and are using it
again to update Docker, you can safely ignore this message.

You may press Ctrl+C now to abort this script.
+ sleep 20
+ sh -c apt-get update -qq >/dev/null
+ sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq ca-certificates curl >/dev/null
+ sh -c install -m 0755 -d /etc/apt/keyrings
+ sh -c curl -fsSL "https://download.docker.com/linux/ubuntu/gpg" -o /etc/apt/keyrings/docker.asc
+ sh -c chmod a+r /etc/apt/keyrings/docker.asc
+ sh -c echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu jammy stable" > /etc/apt/sources.list.d/docker.list
+ sh -c apt-get update -qq >/dev/null
+ sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq docker-ce docker-ce-cli containerd.io docker-compose-plugin docker-ce-rootless-extras docker-buildx-plugin >/dev/null
+ sh -c docker version
Client: Docker Engine - Community
 Version:           27.1.1
 API version:       1.46
 Go version:        go1.21.12
 Git commit:        6312585
 Built:             Tue Jul 23 19:57:01 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          27.1.1
  API version:      1.46 (minimum version 1.24)
  Go version:       go1.21.12
  Git commit:       cc13f95
  Built:            Tue Jul 23 19:57:01 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.19
  GitCommit:        2bf793ef6dc9a18e00cb12efb64355c2c9d5eb41
 runc:
  Version:          1.7.19
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

================================================================================

To run Docker as a non-privileged user, consider setting up the
Docker daemon in rootless mode for your user:

    dockerd-rootless-setuptool.sh install

Visit https://docs.docker.com/go/rootless/ to learn about rootless mode.


To run the Docker daemon as a fully privileged service, but granting non-root
users access, refer to https://docs.docker.com/go/daemon-access/

WARNING: Access to the remote API on a privileged Docker daemon is equivalent
         to root access on the host. Refer to the 'Docker daemon attack surface'
         documentation for details: https://docs.docker.com/go/attack-surface/

================================================================================

sleep 2 # give docker a moment i guess
apt-get -qq -y --no-install-recommends install help2man
apt-get -qq -y --no-install-recommends install man-db
wget -qO /usr/local/bin/sigil https://github.com/gliderlabs/sigil/releases/download/v0.11.0/sigil-linux-amd64
chmod +x /usr/local/bin/sigil
apt-get -qq -y --no-install-recommends install dos2unix
apt-get -qq -y --no-install-recommends install jq
apt-get -qq -y --no-install-recommends install parallel
make -e stack
make[1]: Entering directory '/root/dokku'
Start pulling herokuish from gliderlabs/herokuish:latest-22
docker images | grep gliderlabs/herokuish || docker pull gliderlabs/herokuish:latest-22
gliderlabs/herokuish      latest-22   2b52f88f9979   11 months ago       1.05GB
make[1]: Leaving directory '/root/dokku'
mkdir -p /var/lib/dokku
git describe --tags > /var/lib/dokku/VERSION  2>/dev/null || echo '~master (2024-07-28T22:42+00:00)' > /var/lib/dokku/VERSION
cp dokku /usr/local/bin/dokku
chmod 0755 /usr/local/bin/dokku
make go-build || exit 1
make[1]: Entering directory '/root/dokku'
basedir=/root/dokku; \
for dir in plugins/*; do \
        if [ -e $dir/Makefile ]; then \
                make -e -C $dir build-in-docker || exit $? ;\
        fi ;\
done
make[2]: Entering directory '/root/dokku/plugins/app-json'
rm -rf commands subcommands triggers
find . -xtype l -delete
mkdir -p /tmp/dokku-go-build-cache
docker run --rm \
        -v $PWD/../..:/go/src/github.com/dokku/dokku \
        -v /tmp/dokku-go-build-cache:/root/.cache \
        -v /tmp/dokku-go-mod-mod:/go/pkg/mod \
        -e PLUGIN_NAME=app-json \
        -e GO111MODULE=on \
        -w /go/src/github.com/dokku/dokku/plugins/app-json \
        golang:1.21.6 \
        bash -c "GO_ARGS='' CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GOWORK=off make -j4 build" || exit $?
GOARCH=amd64 go build -mod=readonly -ldflags="-s -w"  -o commands src/commands/commands.go
GOARCH=amd64 go build -mod=readonly -ldflags="-s -w"  -o subcommands/subcommands src/subcommands/subcommands.go
GOARCH=amd64 go build -mod=readonly -ldflags="-s -w"  -o triggers src/triggers/triggers.go
go: updates to go.mod needed, disabled by -mod=readonly; to update it:
        go mod tidy
make: *** [../../common.mk:31: commands] Error 1
make: *** Waiting for unfinished jobs....
go: updates to go.mod needed, disabled by -mod=readonly; to update it:
        go mod tidy
go: updates to go.mod needed, disabled by -mod=readonly; to update it:
        go mod tidy
make: *** [../../common.mk:44: triggers] Error 1
make: *** [../../common.mk:34: subcommands] Error 1
make[2]: *** [../../common.mk:16: build-in-docker] Error 2
make[2]: Leaving directory '/root/dokku/plugins/app-json'
make[1]: *** [Makefile:54: go-build] Error 2
make[1]: Leaving directory '/root/dokku'
make: *** [Makefile:83: copyfiles] Error 1
```

--- a/plugins/scheduler-docker-local/check-deploy
+++ b/plugins/scheduler-docker-local/check-deploy
@@ -153,7 +153,10 @@ trigger-scheduler-docker-local-check-deploy() {
     echo "$content" >"$TMP_APP_JSON_OUTPUT"
   fi
 
-  sudo /usr/bin/docker-container-healthchecker check "$DOKKU_APP_CONTAINER_ID" "${ARG_ARRAY[@]}" || FAILEDCHECKS="$?"
+  local docker_container_healthchecker_path
+  docker_container_healthchecker_path="$(command -v docker-container-healthchecker)"
+
+  sudo "$docker_container_healthchecker_path" check "$DOKKU_APP_CONTAINER_ID" "${ARG_ARRAY[@]}" || FAILEDCHECKS="$?"
 
   if [[ $FAILEDCHECKS -gt 0 ]]; then
     "$DOCKER_BIN" container update --restart=no "$DOKKU_APP_CONTAINER_ID" &>/dev/null || true

--- a/plugins/scheduler-docker-local/install
+++ b/plugins/scheduler-docker-local/install
@@ -16,7 +16,9 @@ trigger-scheduler-docker-local-install() {
 
   rm -f "/etc/sudoers.d/dokku-cron"
 
-  echo "%dokku ALL=(ALL) NOPASSWD:/usr/bin/docker-container-healthchecker" >"/etc/sudoers.d/dokku-docker-container-healthchecker"
+  local docker_container_healthchecker_path
+  docker_container_healthchecker_path="$(command -v docker-container-healthchecker)"
+  echo "%dokku ALL=(ALL) NOPASSWD:$docker_container_healthchecker_path" >"/etc/sudoers.d/dokku-docker-container-healthchecker"
   chmod "0440" "/etc/sudoers.d/dokku-docker-container-healthchecker"
 
   DOKKU_PATH="$(command -v dokku)"

